tests/test_path_follow_3d.h

```diff
@@ -0,0 +1,210 @@
+/*************************************************************************/
+/*  test_path_follow_3d.h                                                */
+/*************************************************************************/
+/*                       This file is part of:                           */
+/*                           GODOT ENGINE                                */
+/*                      https://godotengine.org                          */
+/*************************************************************************/
+/* Copyright (c) 2007-2021 Juan Linietsky, Ariel Manzur.                 */
+/* Copyright (c) 2014-2021 Godot Engine contributors (cf. AUTHORS.md).   */
+/*                                                                       */
+/* Permission is hereby granted, free of charge, to any person obtaining */
+/* a copy of this software and associated documentation files (the       */
+/* "Software"), to deal in the Software without restriction, including   */
+/* without limitation the rights to use, copy, modify, merge, publish,   */
+/* distribute, sublicense, and/or sell copies of the Software, and to    */
+/* permit persons to whom the Software is furnished to do so, subject to */
+/* the following conditions:                                             */
+/*                                                                       */
+/* The above copyright notice and this permission notice shall be        */
+/* included in all copies or substantial portions of the Software.       */
+/*                                                                       */
+/* THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,       */
+/* EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF    */
+/* MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.*/
+/* IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY  */
+/* CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT,  */
+/* TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE     */
+/* SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.                */
+/*************************************************************************/
+
+#ifndef TEST_PATH_FOLLOW_3D_H
+#define TEST_PATH_FOLLOW_3D_H
+
+#include "scene/3d/path_3d.h"
+#include "scene/resources/curve.h"
+
+#include "tests/test_macros.h"
+
+namespace TestPathFollow3D {
+
+TEST_CASE("[PathFollow3D] Sampling with unit offset") {
+	const Ref<Curve3D> &curve = memnew(Curve3D());
+	curve->add_point(Vector3(0, 0, 0));
+	curve->add_point(Vector3(100, 0, 0));
+	curve->add_point(Vector3(100, 100, 0));
+	curve->add_point(Vector3(100, 100, 100));
+	curve->add_point(Vector3(100, 0, 100));
+	const Ref<Path3D> &path = memnew(Path3D());
+	path->set_curve(curve);
+	const Ref<PathFollow3D> &path_follow_3d = memnew(PathFollow3D());
+	path->add_child(path_follow_3d);
```

Comment:
I've only noticed now (and also applies to #46277), those are actually nodes. 😮

So, they are not reference counted (unlike `Curve3D` above). Nodes inherit from `Object`, so they are passed via pointer:

```suggestion
	const Path3D *path = memnew(Path3D);
	path->set_curve(curve);
	const PathFollow3D *path_follow_3d = memnew(PathFollow3D);
	path->add_child(path_follow_3d);
```

That said, I'm kind of surprised why it got compiled and works for nodes, I never thought that regular pointers could be wrapped in `Ref<>`, so this may be a feature I'm simply not aware of.

But for consistency with the rest of the codebase, the suggestion above makes sense. But then you also need to make sure to `memdelete` those after all assertions have passed:

```c++
memdelete(path);
// memdelete(path_follow_3d); // No need to delete this because it's a child of `path`.
```

Also there's no need to specify constructor explicitly `()`, only if you need to pass additional parameters with `memnew`.

The same applies to other test cases and occurences.

Reply:
done

Reply:
will soon commit the same changes on the other PR too.